src/caustic/lenses/kappa_grid.py

```diff
@@ -28,12 +27,32 @@ def __init__(
         mode: str = "fft",
         use_next_fast_len: bool = True,
     ):
+        """Strong lensing with user provided kappa map
+
+        KappaGrid is a class for strong gravitational lensing with a
+        user-provided kappa map. It inherits from the ThinLens class.
+        This class enables the computation of deflection angles and
+        lensing potential by applying the user-provided kappa map to a
+        grid using either Fast Fourier Transform (FFT) or a 2D
+        convolution.
+
+        Attributes:
+            name (str): The name of the KappaGrid object.
+            fov (float): The field of view in arcseconds.
+            n_pix (int): The number of pixels on each side of the grid.
+            cosmology (Cosmology): An instance of the cosmological parameters.
+            z_l (Optional[Tensor]): The redshift of the lens.
+            thx0 (Optional[Tensor]): The x-coordinate of the center of the grid.
+            thy0 (Optional[Tensor]): The y-coordinate of the center of the grid.
+            kappa_map (Optional[Tensor]): A 2D tensor representing the kappa map.
+            kappa_map_shape (Optional[tuple[int, ...]]): The shape of the kappa map.
+            mode (str, optional): The convolution mode for calculating deflection angles and lensing potential.
+                It can be either "fft" (Fast Fourier Transform) or "conv2d" (2D convolution). Default is "fft".
+            use_next_fast_len (bool, optional): If True, adds additional padding to speed up the FFT by calling
+                `scipy.fft.next_fast_len`. The speed boost can be substantial when `n_pix` is prime. Default is True.
```

Comment:
I believe the speedup happens when n_pix is a multiple of some factors that make FFT efficient. I copy here scipy's doc string so that we can make ours reflect theirs better.

SciPy’s FFT algorithms gain their speed by a recursive divide and conquer strategy. This relies on efficient functions for small prime factors of the input length. Thus, the transforms are fastest when using composites of the prime factors handled by the fft implementation. If there are efficient functions for all radices <= n, then the result will be a number x >= target with only prime factors < n. (Also known as n-smooth numbers)